ceph_iscsi_config/lun.py

```diff
@@ -574,6 +602,33 @@ def activate(self):
             if client_err:
                 raise CephiSCSIError(client_err)
 
+    def _erasure_pool_check(self):
+        if not self.is_erasure_pool:
+            return None
+
+        data, err = run_shell_cmd(
+            "ceph -n {client_name} --conf {cephconf} osd metadata --format=json".
+            format(client_name=settings.config.cluster_client_name,
+                   cephconf=settings.config.cephconf))
+        if err:
+            self.logger.error("Cannot get the objectstore type")
+            return err
+        store_type = json.loads(data)[0]['osd_objectstore']
+        self.logger.debug(f"pool ({self.pool}) objectstore type is ({store_type})")
+        if store_type not in ['bluestore', 'filestore']:
+            self.logger.error("Only bluestore/filestore objectstores allowed for erasure pool")
```

Comment:
This check is bogus because one can have a cluster with OSDs backed by different stores.  Querying just one OSD isn't going to do it.

But then I don't understand the need to check `osd_objectstore` at all.  Why ensuring that EC overwrites are enabled isn't enough?

Reply:
> This check is bogus because one can have a cluster with OSDs backed by different stores. Querying just one OSD isn't going to do it.
> 
Okay.

> But then I don't understand the need to check `osd_objectstore` at all. Why ensuring that EC overwrites are enabled isn't enough?

From https://docs.ceph.com/en/latest/rados/operations/erasure-code/#erasure-coding-with-overwrites, I think maybe we should only allow the bluestore ?

Reply:
I don't think we should be trying to limit it beyond what is allowed by librbd.  When creating an image with a separate data pool, librbd ensures that the specified pool supports overwrites.

The user wouldn't be able to set `allow_ec_overwrites` on a filestore-backed pool without also messing with `mon_debug_no_require_bluestore_for_ec_overwrites`.  If they do that, they are on their own ;)